### PR TITLE
✨ Script to build kubectl image

### PR DESCRIPTION
**Pull request by francostellari**
## Summary

Script to build kubectl image

## Related issue(s)

Fixes #

### CODE REVIEW

File: hack/Dockerfile.kubectl [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Copyright YEAR The KubeStellar Authors.
```
Comment (waltforme) [inline hack/Dockerfile.kubectl:1]: Sounds like any `YEAR`?

File: hack/Dockerfile.kubectl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# RUN microdnf install -y findutils
+
+USER 1001
+
+ENTRYPOINT [ "kubectl" ]
+CMD [ "--help" ]
+
+
+# https://github.com/AICoE/kubectl-container/blob/master/Dockerfile
+# https://gist.github.com/rponte/fdc0724dd984088606b0
```
Comment (waltforme) [inline hack/Dockerfile.kubectl:43]: Some comment about this red symbol: https://github.com/kubestellar/kubestellar/pull/1967#discussion_r1541741908.

File: hack/build-kubectl-image.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+docker buildx create --name kubestellar --platform ${platform} --use
+
+docker buildx build . \
+    --push \
+    --platform ${platform} \
+    --tag quay.io/kubestellar/kubectl:${version#v} \
+    --build-arg version=${version#v} \
+    --tag quay.io/kubestellar/kubectl:${version#v} \
+    -f /vagrant/Dockerfile.kubectl
```
Comment (waltforme) [inline hack/build-kubectl-image.sh:81]: I don't understand the `vagrant` here.

File: hack/Dockerfile.kubectl (hunk 1 above)
Comment (francostellari) [reply hack/Dockerfile.kubectl]: @waltforme thank you for catching it. Fixed in the commit.

File: hack/Dockerfile.kubectl (hunk 2 above)
Comment (francostellari) [reply hack/Dockerfile.kubectl:43]: Removed, I used as abase

File: hack/Dockerfile.kubectl [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
```
Comment (waltforme) [inline hack/Dockerfile.kubectl:43]: BTW, are these two references intentionally kept in this dockerfile?

File: hack/Dockerfile.kubectl [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    curl \
+	-L https://github.com/stedolan/jq/releases/latest/download/jq-${TARGETOS}-${TARGETARCH} \
+	-o /usr/bin/jq && \
+	chmod +x /usr/bin/jq && \
+    curl \
+	-L https://github.com/mikefarah/yq/releases/latest/download/yq_${TARGETOS}_${TARGETARCH} \
+	-o /usr/bin/yq && \
+	chmod +x /usr/bin/yq
+
+# RUN microdnf install -y findutils
```
Comment (waltforme) [inline hack/Dockerfile.kubectl:34]: Maybe we just remove this if we don't need it?

File: hack/build-kubectl-image.sh (hunk 3 above)
Comment (francostellari) [reply hack/build-kubectl-image.sh:81]: leftover from testing, fixed

File: hack/Dockerfile.kubectl (hunk 4 above)
Comment (francostellari) [reply hack/Dockerfile.kubectl:43]: removed

File: hack/Dockerfile.kubectl (hunk 5 above)
Comment (francostellari) [reply hack/Dockerfile.kubectl:34]: yes, I commented to check if we needed it and forgot to remove it
removed it now

**Pull request merged**